### PR TITLE
Port Changes Made to Yoga Since the Fork

### PR DESCRIPTION
**Pull request by Amdrel**
~Depends on PR: https://github.com/thisdot/typeflex/pull/2~

### CODE REVIEW

File: src/event.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        measureCallbackReasonsCount: LayoutPassCounts = [0, 0, 0, 0, 0, 0, 0, 0],
+    ) {
+        this.layouts = layouts;
+        this.measures = measures;
+        this.maxMeasureCache = maxMeasureCache;
+        this.cachedLayouts = cachedLayouts;
+        this.cachedMeasures = cachedMeasures;
+        this.measureCallbacks = measureCallbacks;
+        this.measureCallbackReasonsCount = measureCallbackReasonsCount;
+    }
```
Comment (ktrz) [inline src/event.ts]: just for code improving purposes you could make it much smaller by using a `public` keyword in constructor
```suggestion
    constructor(
        public layouts = 0,
        public measures = 0,
        public maxMeasureCache = 0,
        public cachedLayouts = 0,
        public cachedMeasures = 0,
        public measureCallbacks = 0,
        public measureCallbackReasonsCount: LayoutPassCounts = [0, 0, 0, 0, 0, 0, 0, 0],
    ) {}
```

File: src/event.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+const subscribers: Subscribers = [];
+
+function push(newHead: Subscriber): Subscriber {
+    let oldHead: Subscriber = null;
+    if (subscribers.length > 0) {
+        oldHead = subscribers[subscribers.length - 1];
+    }
+    subscribers.push(newHead);
+    return oldHead;
+}
```
Comment (ktrz) [inline src/event.ts:163]: is it a deviation from upstream? In the [upstream code](https://github.com/facebook/yoga/blob/main/yoga/event/event.cpp#L50) I've found there is a `do .. while` loop here

Comment (Amdrel) [reply src/event.ts:163]: It is a deviation as indicated by the comment above on line 164. The original implementation uses a reversed linked list and atomics for thread safety, and that is not needed in a TypeScript implementation. For our case I thought a simple array made more sense.

File: src/api.ts [hunk 3]
```diff
@@ -723,11 +734,23 @@ export class Node {
...
+    }
+
+    setIsReferenceBaseline(isReferenceBaseline: boolean): void {
+        YGNodeSetIsReferenceBaseline(this.node, isReferenceBaseline);
+    }
 }
+
+// export function getInstanceCount(): number {
+//     return YGNodeGetInstanceCount();
+// }
```
Comment (ktrz) [inline src/api.ts]: can we remove this commented-out code?

**Review by ktrz — APPROVED** [review]
just one small comment
otherwise LGTM!

Kudos to you for this huge effort!

**Pull request merged**